opencti-platform/opencti-front/src/private/components/workspaces/dashboards/Dashboard.jsx

```diff
@@ -140,7 +133,28 @@ const DashboardComponent = ({ workspace, noToolbar }) => {
     }
     saveManifest(newManifest);
   };
+  const getMaxY = () => {
+    const maxY = Object.values(manifest.widgets).reduce(
+      (max, n) => (n.layout.y > max ? n.layout.y : max),
+      0,
+    );
+    return maxY;
+  };
+  const getMaxX = () => {
+    const y = getMaxY();
+    const maxX = Object.values(manifest.widgets).filter((n) => n.layout.y === y).reduce(
+      (max, n) => (n.layout.x > max ? n.layout.x : max),
+      0,
+    );
+    return maxX + 4;
```

Comment:
The hard coded `4` works only when the widget on the left of the one added is 4 cell wide.
If the widget is wider then the new goes below it. If it is narrower there is a space between both widgets.

I think we need to use `layout.w` in the reduce function to find correct maxX

Reply:
I added screen captures but they disappeared 😒

Reply:
Narrower
![image](https://github.com/OpenCTI-Platform/opencti/assets/33682259/8ee26bcb-1f0e-4623-bc29-609f68febedb)

Wider
![image](https://github.com/OpenCTI-Platform/opencti/assets/33682259/7c02411d-7201-4758-9384-afc10337366e)

New line
![image](https://github.com/OpenCTI-Platform/opencti/assets/33682259/26a64ab0-0891-4fcc-853c-300690ae8b3d)